packages/ui/src/app/components/ConnectionStatusDot.tsx

```diff
@@ -18,16 +18,16 @@ interface ConnectionStatusDotProps {
 
 export const ConnectionStatusDot = ({ onlyPerformance = false, className }: ConnectionStatusDotProps) => {
   const { api, connectionState, qnConnectionState } = useApi()
-  const { queryNodeState } = useQueryNodeStateSubscription()
+  const { queryNodeState } = useQueryNodeStateSubscription({ shouldResubscribe: true })
   const header = useObservable(() => api?.rpc.chain.subscribeNewHeads(), [api?.isConnected])
+  const [isQnLate, setIsQnLate] = useState(false)
 
-  const isQnLate = useMemo(() => {
+  useEffect(() => {
     if (queryNodeState && header) {
-      return (
-        (header.toJSON().number as number) - Number(queryNodeState.indexerHead) > MAX_INDEXER_BLOCKS_BEHIND_NODE_HEAD
-      )
+      const processorDelay = Number(header.number) - Number(queryNodeState.indexerHead)
+      setIsQnLate(processorDelay > MAX_INDEXER_BLOCKS_BEHIND_NODE_HEAD)
```

Comment:
I assume `queryNodeState` is updated on every processed block which ideally happens very soon after block production except the node is under load / syncing. Should we allow a delay of a few blocks or would that defeat the purpose?

Reply:
Yes `MAX_INDEXER_BLOCKS_BEHIND_NODE_HEAD` is the allowed delay currently 20 blocks